test/Chirp.Infrastructure.Tests/InMemoryDatabaseUnitTest.cs

```diff
@@ -0,0 +1,89 @@
+using Microsoft.Data.Sqlite;
+using Microsoft.EntityFrameworkCore;
+using FluentAssertions;
+public class InMemoryDatabaseTest
+{
+    private readonly SqliteConnection? _connection; //The connection to the database
+    private readonly ChirpDBContext _context; //The context of the database
+    private readonly CheepRepository repository; //The repository of the memory database
+    private readonly CheepRepository ExistingRepository; //The repository of the existing database
+
+    // private readonly CheepDTO cheep;
+    public InMemoryDatabaseTest()
+    {
+        //Arrrange 
+
+        //Creates a database in memory - Makkes connection string before opening the connection
+        var builder = new DbContextOptionsBuilder<ChirpDBContext>();
+        builder.UseSqlite("Filename=:memory:");
+        ChirpDBContext context = new(builder.Options);
+        _connection = context.Database.GetDbConnection() as SqliteConnection;
+        if (_connection != null) //Takes care of the null exception
+        {
+            _connection.Open();
+        }
+        context.Database.EnsureCreated();
+
+        //Creates a cheep and author to add to the database
```

Comment:
I suggest that we move this out, into each test, because, in my opinion, this makes the test cases less readable and more prone to erros, if more test were to be added, or existings tests were to be changed.
I myself, used som time figuring out what exactly was tested in the test, since the data is added in the constructor already.

Reply:
As long as the test in themself dont need a unique database, they can work in the same, which makes the script more readable. It was unnessesay filling before it were moved to the constructor